spec/system/cookies_consent_spec.rb

```diff
@@ -0,0 +1,162 @@
+require "rails_helper"
+
+describe "Cookies consent" do
+  before do
+    Setting["feature.cookies_consent"] = true
+    script = "$('body').append('Running third party script');"
+    create(:cookies_vendor, name: "Third party", cookie: "third_party", script: script)
+  end
+
+  context "Banner consent" do
+    scenario "Hides the banner when accept essential cookies and for consecutive visits" do
+      visit root_path
+
+      expect(cookie_by_name("cookies_consent_v1")).to be nil
+      expect(cookie_by_name("third_party")).to be nil
+      expect(page).not_to have_content "Running third party script"
+
+      within ".cookies-consent-banner" do
+        click_button "Accept essential cookies"
+      end
+
+      expect(cookie_by_name("cookies_consent_v1")[:value]).to eq "essential"
+      expect(cookie_by_name("third_party")[:value]).to eq "false"
+      expect(page).not_to have_css ".cookies-consent-banner"
```

Comment:
```suggestion
      expect(page).not_to have_content "Cookies policy"
```

This way we test the page from the user's point of view.